go/actions/test/action.yml

```diff
@@ -0,0 +1,76 @@
+name: Test go extractor
+description: Run build, QL tests and optionally basic code sanity checks (formatting and generation)
+inputs:
+  go-version:
+    description: Which Go version to use for running the tests
+    required: false
+    default: ~1.22.0
+  run-code-checks:
+    description: Whether to run formatting, code and qhelp generation checks
+    required: false
+    default: false
+runs:
+  using: composite
+  steps:
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: ${{ inputs.go-version }}
+        cache: false
+      id: go
+
+    - name: Set up CodeQL CLI
+      uses: ./.github/actions/fetch-codeql
+
+    - name: Enable problem matchers in repository
+      shell: bash
+      run: 'find .github/problem-matchers -name \*.json -exec echo "::add-matcher::{}" \;'
+
+    - name: Build
+      shell: bash
+      run: |
+        bazel run go:create-extractor-pack
+
+    - name: Check that all Go code is autoformatted
+      if: inputs.run-code-checks == 'true'
+      shell: bash
+      run: |
+          cd go
+          make check-formatting
+
+    - name: Check checked-in generated code
+      if: inputs.run-code-checks == 'true'
+      shell: bash
+      run: |
+        bazel run go:gen
+        git add .
+        git diff --exit-code HEAD || (
+          echo "please run bazel run //go:gen"
+          exit 1
+        )
+
+    - name: Compile qhelp files to markdown
+      if: inputs.run-code-checks == 'true'
```

Comment:
Same as above

Reply:
yeah, this on the other hand might make sense to render independent... but maybe we could do that with a separate parallel job?

Reply:
A separate parallel job would work, too. The downside to doing that may be that it requires the overhead of checking out the repo on a separate runner. It's a question of whether we care about that overhead or not. I have no strong opinion.

Reply:
I ended up keeping it here, as there was also the `go` setup being required, which I prefer having shared here with one place where the go version should be updated.